docs/plugin/messaging.md

```diff
@@ -0,0 +1,601 @@
+
+```c++
+void AnalyseAndReportDispatchedJobs()
+{
+    _lock.Lock();
+
+    if (_dispatchedJobList.size() > 0 && IsActive()) {
+        for (auto &job : _dispatchedJobList) {
+            ++job.ReportRunCount;
+            REPORT_OUTOFBOUNDS_WARNING_EX(WarningReporting::JobActiveForTooLong, job.CallSign.c_str(),
+            static_cast<uint32_t>((Time::Now().Ticks() - job.DispatchedTime) / Time::TicksPerMillisecond));
+        }
+    }
+    _lock.Unlock();
+}
+```
+
+# MessageControl plugin
+
+Now that we have explored the concepts of logging, tracing, and warning reporting in theory and their implementation within Thunder, we can delve into the significance of the `MessageControl` plugin. This plugin plays a vital role as it manages all message types seamlessly. In the subsequent sections, we will provide a more detailed explanation of how the plugin operates and the specific steps it undertakes. However, it is important to highlight that the plugin not only consolidates various message types but also offers the flexibility to direct these messages to different specific outputs. These outputs include the Console, Syslog, a file, or even a Network Stream through UDP.
+
+## Content of the message
+
+Before examining the workings of the plugin, it is crucial to provide a brief overview of the components that make up a message, as they directly impact the message processing. A message consists of three main parameters: module, category, and the actual content of the message. Understanding these parameters is essential for effective message handling.
+
+### Categories of each message type
+
+When it comes to tracing, the module parameter represents the name of the plugin responsible for sending the message, for example `Plugin_Cobalt`. However, when the `SYSLOG` macro is used, the module name will simply be `Syslog`.
+
+Choosing the appropriate category is vital when working with macros. The category can be described as the class that either formats the message string or best represents the nature of the message. While it is possible to create custom categories, which will be explained in the following paragraph, it is generally recommended to utilize the predefined categories available in Thunder to avoid redundancy. Thunder already offers six default categories for tracing:
+
+-   Text
+
+-   Initialization
+
+-   Information
+
+-   Warning
+
+-   Error
+
+-   Fatal
+
+Each of them is created in `Thunder/Source/messaging/TraceCategories.h` using the `DEFINE_MESSAGING_CATEGORY` macro:
+
+```c++
+DEFINE_MESSAGING_CATEGORY(Messaging::BaseCategoryType<Core::Messaging::Metadata::type::TRACING>, Text);
+```
+
+As you can see in the code, these classes inherit from a templated class called `BaseCategoryType`. The association of these categories with tracing is established by passing the `TRACING` metadata type as a template parameter. Similarly, logging includes several categories, such as:
+
+-   Startup
```

Comment:
Sounds like a good idea. Let's have a configuration section in this document like you've suggested in your table of contents. I'll move the part of config documentation related to messaging to here, and also change it to match the current changes, since I can see that it is not entirely correct now.